common-http/src/main/java/io/cdap/common/http/HttpRequestConfig.java

```diff
@@ -79,4 +84,8 @@ public boolean isVerifySSLCert() {
   public int getFixedLengthStreamingThreshold() {
     return fixedLengthStreamingThreshold;
   }
+
+  public boolean consumeResponseStream() {
```

Comment:
Why we need this? You can make the HttpResponse takes a Supplier for body byte[] so that data will be read (and memorized) upon `getResponseBodyXXX` is called.

Reply:
fixed